src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -106,5 +107,17 @@ public void doPut(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doDelete(HttpServletRequest request, HttpServletResponse response)
-      throws IOException {}
+      throws IOException {
+    String recipeID = request.getParameter("recipeID");
+    String commentID = request.getParameter("commentID");
+    String json;
+
+    if (recipeID == null || commentID == null) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+
+    ApiFuture<WriteResult> writeResult =
```

Comment:
I think it would be reasonable to send the response once the delete has actually gone through, so calling `writeResult.get()` would be a good idea.